### PR TITLE
add link training status, prbs config, an status in saiport.h

### PR DESCRIPTION
**Pull request by apogedik**
Broadcom proposal to add link training failure status

### CODE REVIEW

**Comment by kcudnik** [comment]
please address compilation issues: https://sonic-jenkins.westus2.cloudapp.azure.com/job/sai-meta-pr-build/1826/console

File: inc/saiport.h [hunk 1]
```diff
@@ -1293,7 +1312,14 @@ typedef enum _sai_port_attr_t
      * @default internal
      */
     SAI_PORT_ATTR_PORT_SERDES_ID,
-
+    
+    /** @brief Link training failure status and error codes
+     * @type sai_port_link_training_fail_status_t
+     * @flags READ_ONLY
+     * @default internal
```
Comment (kcudnik) [inline inc/saiport.h]: default is not required if attribute is read only

File: inc/saiport.h [hunk 2]
```diff
@@ -1293,7 +1312,14 @@ typedef enum _sai_port_attr_t
      * @default internal
      */
     SAI_PORT_ATTR_PORT_SERDES_ID,
-
+    
+    /** @brief Link training failure status and error codes
```
Comment (kcudnik) [inline inc/saiport.h]: move @brief to next line, and add empty line before type, please look at style on other attributes

File: inc/saiport.h (hunk 1 above)
Comment (apogedik) [reply inc/saiport.h]: done

File: inc/saiport.h (hunk 2 above)
Comment (apogedik) [reply inc/saiport.h]: done

**Comment by kcudnik** [comment]
you have white spaces on line endings, please remove, and add SNR  to acronyms file

**Comment by kcudnik** [comment]
please address compilation errors

File: inc/saiport.h [hunk 3]
```diff
@@ -2230,6 +2322,43 @@ typedef sai_status_t (*sai_clear_port_pool_stats_fn)(
         _In_ uint32_t number_of_counters,
         _In_ const sai_stat_id_t *counter_ids);
 
+/**
+ * @brief Attribute data for #SAI_PORT_SERDES_ATTR_TX_FIR_TAP_MODE
+ *
+ * TX FIR Tap modes for NRZ and PAM4
```
Comment (apogedik) [inline inc/saiport.h]: I keep getting "WARNING: Word 'NRZ' is misspelled or is acronym, add to acronyms.txt? saiport.h 2328: * TX FIR Tap modes for NRZ and PAM4"  warning.  NRZ is already listed in the acronyms file.  Why is this warning?  How do we fix it?

**Pull request merged**